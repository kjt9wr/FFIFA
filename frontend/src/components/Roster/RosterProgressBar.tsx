import { Progress } from "reactstrap";

interface RosterProgressBarProps {
  keepPrice: number;
  taxLine: number;
  maxCap: number;
}

const RosterProgressBar = (props: RosterProgressBarProps) => {
  const { keepPrice, taxLine, maxCap } = props;
  const isOffender = keepPrice > taxLine;
  return (
    <Progress
      multi
      style={{
        height: "60px",
      }}
    >
      <Progress
        bar
        color="success"
        value={isOffender ? 55 : (keepPrice / maxCap) * 100}
      >
        Under the Tax
      </Progress>
      <Progress
        bar
        color="warning"
        value={isOffender ? ((keepPrice - taxLine) / maxCap) * 100 : 0}
      >
        $ Over
      </Progress>
      <Progress
        bar
        color="danger"
        value={isOffender ? ((keepPrice - taxLine) / maxCap) * 100 : 0}
      >
        $ Penalty
      </Progress>
    </Progress>
  );
};

export default RosterProgressBar;
