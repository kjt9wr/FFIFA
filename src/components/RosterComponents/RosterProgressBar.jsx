import React from "react";
import { Progress } from "reactstrap";

const RosterProgressBar = (props) => {
  const { isOffender, keepPrice, TAX_LINE, MAX_CAP } = props;
  return (
    <Progress
      multi
      style={{
        height: "60px",
      }}
    >
      <Progress
        bar
        color="success"
        value={isOffender ? 55 : (keepPrice / MAX_CAP) * 100}
      >
        Under the Tax
      </Progress>
      <Progress
        bar
        color="warning"
        value={isOffender ? ((keepPrice - TAX_LINE) / MAX_CAP) * 100 : 0}
      >
        $ Over
      </Progress>
      <Progress
        bar
        color="danger"
        value={isOffender ? ((keepPrice - TAX_LINE) / MAX_CAP) * 100 : 0}
      >
        $ Penalty
      </Progress>
    </Progress>
  );
};

export default RosterProgressBar;
