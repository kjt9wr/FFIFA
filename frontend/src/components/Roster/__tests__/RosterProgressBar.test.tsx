import { render } from "@testing-library/react";
import RosterProgressBar from "../RosterProgressBar";

describe("roster progress bar", () => {
  it("renders successfully for offender", () => {
    const view = render(
      <RosterProgressBar keepPrice={130} taxLine={110} maxCap={200} />
    );
    expect(view).toMatchSnapshot();
  });

  it("renders successfully for non-offender", () => {
    const view = render(
      <RosterProgressBar keepPrice={100} taxLine={110} maxCap={200} />
    );
    expect(view).toMatchSnapshot();
  });
});
