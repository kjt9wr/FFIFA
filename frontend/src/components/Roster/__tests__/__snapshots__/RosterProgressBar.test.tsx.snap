// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`roster progress bar renders successfully for non-offender 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="progress"
        style="height: 60px;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="50"
          class="progress-bar success"
          role="progressbar"
          style="width: 50%;"
        >
          Under the Tax
        </div>
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="progress-bar warning"
          role="progressbar"
          style="width: 0%;"
        >
          $
          -10
        </div>
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="progress-bar penalty"
          role="progressbar"
          style="width: 0%;"
        >
          $
          -10
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="progress"
      style="height: 60px;"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="progress-bar success"
        role="progressbar"
        style="width: 50%;"
      >
        Under the Tax
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="progress-bar warning"
        role="progressbar"
        style="width: 0%;"
      >
        $
        -10
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="progress-bar penalty"
        role="progressbar"
        style="width: 0%;"
      >
        $
        -10
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`roster progress bar renders successfully for offender 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="progress"
        style="height: 60px;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="55"
          class="progress-bar success"
          role="progressbar"
          style="width: 55.00000000000001%;"
        >
          Under the Tax
        </div>
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="10"
          class="progress-bar warning"
          role="progressbar"
          style="width: 10%;"
        >
          $
          20
        </div>
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="10"
          class="progress-bar penalty"
          role="progressbar"
          style="width: 10%;"
        >
          $
          20
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="progress"
      style="height: 60px;"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="55"
        class="progress-bar success"
        role="progressbar"
        style="width: 55.00000000000001%;"
      >
        Under the Tax
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="progress-bar warning"
        role="progressbar"
        style="width: 10%;"
      >
        $
        20
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="progress-bar penalty"
        role="progressbar"
        style="width: 10%;"
      >
        $
        20
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
